hooks/index.ts

```diff
@@ -21,3 +27,51 @@ export async function caseStudyFullInfo() {
   const caseStudies: CaseStudyType[] = await getCaseStudy();
   return caseStudies;
 }
+
+export async function getHomePageData() {
+  const results = await Promise.allSettled([
+    caseStudyInfoHomepage(),
+    getTestimonials(),
+    getWorkExperiences(),
+    getServicesProvided(),
+    getMySkills(),
+  ]);
+
+  const [
+    caseStudiesResult,
+    testimonialsResult,
+    workExperienceListResult,
+    servicesResult,
+    mySkillsResult,
+  ] = results;
+
+  const caseStudies =
+    caseStudiesResult.status === "fulfilled" ? caseStudiesResult.value : [];
+  const testimonials =
+    testimonialsResult.status === "fulfilled" ? testimonialsResult.value : [];
+  const workExperienceList =
+    workExperienceListResult.status === "fulfilled"
+      ? workExperienceListResult.value
+      : [];
+  const services =
+    servicesResult.status === "fulfilled" ? servicesResult.value : [];
+  const mySkills =
+    mySkillsResult.status === "fulfilled" ? mySkillsResult.value : [];
+
+  const errors = results
+    .filter((result) => result.status === "rejected")
+    .map((result) => (result as PromiseRejectedResult).reason);
+
+  if (errors.length) {
+    console.error("Some errors occurred during data fetching:", errors);
+  }
```

Comment:
Great job! I am very proud of your approach to this suggestion. Not only did you implement the changes with promise.all(), you took it a step further to implement allSettled(). Again, Great work!!! I will include the main difference between the two below. Also, I am including another approach for your error checking. You might need to adjust it but this approach should log the specific errors while running the map function first... 

- Use Promise.all when you need all promises to succeed and want to fail fast if any of the promises reject.
- Use Promise.allSettled when you want to know the outcome of each promise and don't need all of them to succeed.


```
  // Log specific errors
  const errors = results
    .map((result, index) => {
      if (result.status === "rejected") {
        const reason = (result as PromiseRejectedResult).reason;
        console.error(`Error fetching data for index ${index}:`, reason);
        return reason;
      }
      return null;
    })
    .filter((e) => e !== null);
```